spec/lib/server/subscriber_spec.rb

```diff
@@ -4,63 +4,59 @@
   include EM::TestHelper
 
   let(:channel_key)     { '/bears/are/mean' }
+  let(:chan_sub)        { Firehose::Server::ChannelSubscription.new(channel_key) }
   let(:subscriber)      { Firehose::Server::Subscriber.new(Firehose::Server.redis.connection) }
   let(:dummy_subscriber){ Firehose::Server::Subscriber.new(double('redis', :pubsub => double('pubsub', :subscribe => EM::DefaultDeferrable.new, :on => nil))) }
   let(:message)         { 'Raaaarrrrrr!!!!' }
   let(:publisher)       { Firehose::Server::Publisher.new }
 
   describe "#subscribe" do
     it "adds the deferrable to the subscriptions hash" do
-      deferrable = EM::DefaultDeferrable.new
-      dummy_subscriber.subscribe(channel_key, deferrable)
-      expect(dummy_subscriber.send(:subscriptions)[channel_key]).to eql([deferrable])
-    end
-
-    it "calls succeed on the deferrable when a message is published" do
-      deferrable = EM::DefaultDeferrable.new
-      deferrable.callback do |messages|
-        msg = messages.first
-        expect(msg.payload).to eql(message)
-        expect(msg.sequence).to eql(1) # The publisher is fresh, so the sequence ID will be 1.
+      em do
+        dummy_subscriber.subscribe(chan_sub)
+        expect(dummy_subscriber.send(:subscriptions)[channel_key]).to eql([chan_sub])
         em.next_tick { em.stop }
       end
+    end
 
+    it "calls process_messages on the deferrable when a message is published" do
       em do
-        subscriber.subscribe(channel_key, deferrable)
-        publisher.publish(channel_key, message)
+        expect(chan_sub).to receive(:process_messages)#.with([Firehose::Server::Message.new(message, 1)])
```

Comment:
This is failing in CI for Ruby 2.2.5: https://travis-ci.org/firehoseio/firehose/jobs/142329584#L279

I can't reproduce the problem locally (I only tried version 2.2.2).